src/Application/main.cpp

```diff
@@ -48,44 +48,27 @@ void usage(char **argv) {
 
 int main(int argc, char* argv[])
 {
-	loguru::init(argc, argv);
-	LOG_F(INFO, "Space Ring Things - Game Server");
-
-    std::string     strBrokerURI = "tcp://127.0.0.1:5672";
-    std::string     strCommandInDestinationURI = "COMMAND.IN";
-    std::string     strGameEventOutDestinationURI = "GAME.EVENT.OUT";
-    std::string     strServerSleepCycle = "1500";
-
-    LOG_F(INFO, "Starting...");
-    for (int i = 1; i < argc; ++i)
+    // First see if they are looking to display usage via --help as first argument
+    if (0 == strcmp(argv[0], "--help"))
     {
-        if (0 == strcmp(argv[i], "--help"))
-        {
-            usage(argv);
-            exit(0);
-        }
-        else if (0 == strcmp(argv[i], "--broker-uri"))
-        {
-            // TODO: Process -> Usage Error checking on arg
-            strBrokerURI = argv[++i];
-        }
-        else if (0 == strcmp(argv[i], "--sleep-cycle"))
-        {
-            strServerSleepCycle = argv[++i];
-        }
+        usage(argv);
+        exit(0);
     }
 
-    Configuration::Instance().BrokerURI = strBrokerURI;
-    Configuration::Instance().ServerSleepCycle = strtol(strServerSleepCycle.c_str(), nullptr, 0);
+	loguru::init(argc, argv);
+	LOG_F(INFO, "Space Ring Things - Game Server");
+
+	Config::init(argc, argv);
```

Comment:
This splits the initialization in what for c++ is an odd way. Using the established norm for creating the singleton that returns a ref to the static single instance first and then calling init on the reference would likely be better form.

As well the naming using init doesn't follow established code within the codebase rather following the ::init from loguru perhaps that is a 3rd party lib. The naming should be Initialize with a capital I as method names tend to be upper camel case.

Reply:
Yeah this one is weird, because it's a singleton that has to be initialized with `argv[]` So that initization has to take place somehow, and it can't be a constructor because it's a singleton.  I went a head and called the init() function on the instance rather than calling it as a static method of the class like you suggested.